pkg/kustomize/apply.go

```diff
@@ -66,7 +68,15 @@ func (s *Kustomizer) ApplyKustomizationPath(path string) {
 }
 
 func ApplyKustomizationWithRetries(kustomization string, kubeconfig string) error {
-	return wait.Poll(retryInterval, retryTimeout, func() (bool, error) {
+	retryCount := 0
+
+	return wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+		retryCount++
+		if retryCount > retryMaxCount {
+			retryError := fmt.Errorf("reached maximum of %v retries", retryMaxCount)
+			klog.Infof("Applying kustomization failed: %v", retryError)
```

Comment:
Small nit, this is already logged in line 61, but up to you :)

Reply:
Agreed. Let's make it even simpler.